packages/crawler/src/page-handler/page-navigation-hooks.ts

```diff
@@ -30,16 +30,18 @@ export class PageNavigationHooks {
         response: Puppeteer.HTTPResponse,
         onNavigationError: (browserError: BrowserError) => Promise<void> = () => Promise.resolve(),
     ): Promise<Partial<void>> {
-        if (isNil(response)) {
+        const pageUrl = page.url();
+        const isHashUrl = !isNil(pageUrl) && pageUrl.includes('#');
+        if (response === undefined || (!isHashUrl && response === null)) {
             await onNavigationError({
                 errorType: 'NavigationError',
                 message: 'Unable to get a page response from the browser.',
                 stack: new Error().stack,
             });
         }
 
-        // Validate HTTP response
-        const responseError = this.pageResponseProcessor.getResponseError(response);
+        // Validate HTTP response, hashUrls will have null response as crawler treats it same page load
```

Comment:
Suggestion, since we're in the crawler package, it's probably appropriate to be specific about the source of this behavior
```suggestion
        // Validate HTTP response
        // hashUrls will have null response as crawlee's page.goto treats it as same page load
```

Reply:
Done